src/fernet.erl

```diff
@@ -207,59 +269,51 @@ decode_key_test() ->
 %]
 % 1985-10-26T01:20:00-07:00 == 499162800 Seconds since the epoch
 %
-generate_token_test() ->
-  Tok = generate_token("hello", <<0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15>>, 499162800, decode_key("cw_0x689RpI-jtRR7oE8h_eQsKImvJapLeSbXpwF4e4=")),
-  ?assertEqual(decode_token("gAAAAAAdwJ6wAAECAwQFBgcICQoLDA0ODy021cpGVWKZ_eEwCGM4BLLF_5CV9dOPmrhuVUPgJobwOz7JcbmrR64jVmpU4IwqDA=="),
-               decode_token(Tok)).
+generate_token_test_() ->
```

Comment:
Ah yes. You'll notice that instead of having

```erlang```
?assertEqual(A, B),
?assertEqual(C, D).
```
to end the test, I have instead:

```erlang
[?_assertEqual(A,B),
 ?_assertEqual(C,D)].
```
This, instead of being an assertion, is a *generator*, which produces a test case and returns it as a function. i.e. it's pretty much equivalent to 

```erlang
[fun() -> ?assertEqual(A,B) end,
 fun() -> ?assertEqual(C,D) end].
```
Which is a list of functions containing unit tests. EUnit generall perceives unit tests as all functions that end with `_test` and take 0 arguments. To tell Eunit that what I have is a function that is gonna return is a list of tests, I have to declare it as a generator. So instead of having `_test()`, I have `_test_()` as a suffix, and EUnit interprets that to say "I'm gonna get a list of unit tests, run them all".

The final difference is that instead of EUnit reporting `generate_token_test()` as a single test (with two assertions), it is going to instead report two independent tests.